diff --git a/sdk/sdk/src/arch/linux/net_socket.cpp b/sdk/sdk/src/arch/linux/net_socket.cpp
index 0124a13..d270844 100644
--- a/sdk/sdk/src/arch/linux/net_socket.cpp
+++ b/sdk/sdk/src/arch/linux/net_socket.cpp
@@ -167,7 +167,7 @@ u_result SocketAddress::getAddressAsString(char * buffer, size_t buffersize) con
 
         break;
     }
-    return ans<=0?RESULT_OPERATION_FAIL:RESULT_OK;
+    return *ans<=0?RESULT_OPERATION_FAIL:RESULT_OK;
 }
 
 
diff --git a/sdk/sdk/src/hal/event.h b/sdk/sdk/src/hal/event.h
index 5e05234..5fba726 100644
--- a/sdk/sdk/src/hal/event.h
+++ b/sdk/sdk/src/hal/event.h
@@ -95,7 +95,7 @@ public:
         }
     }
     
-    unsigned long wait( unsigned long timeout = 0xFFFFFFFF )
+    long wait( unsigned long timeout = 0xFFFFFFFF )
     {
 #ifdef _WIN32
         switch (WaitForSingleObject(_event, timeout==0xFFFFFFF?INFINITE:(DWORD)timeout))
